main.go

```diff
@@ -118,21 +147,29 @@ func siteHandler(w http.ResponseWriter, r *http.Request) {
 			return
 		}
 
-		if album, err := site.GetAlbumForPath(path); err != nil {
-			// If the path doesn't have a / at it's end, try to see if we can find an album after adding the /
-			if path[len(path)-1] != '/' {
-				if _, err := site.GetAlbumForPath(path + "/"); err == nil {
-					// If we did find it, redirect user to there
-					http.Redirect(w, r, path+"/", http.StatusMovedPermanently)
-					return
-				}
+		album, err := site.GetAlbumForPath(path)
+		if err != nil {
+			i := strings.LastIndex(path, "/") + 1
+			albumPath := path[:i]
+			slug := path[i:]
+
+			album, err = site.GetAlbumForPath(albumPath)
+			if err != nil {
+				w.WriteHeader(http.StatusNotFound)
+				w.Write([]byte(err.Error()))
+				return
 			}
-			w.WriteHeader(http.StatusNotFound)
-			w.Write([]byte(err.Error()))
+
+			//TODO make sure image exists -- use HeadObject. Redirect to album if DNE
```

Comment:
Maybe we could knock this one out as well before merging this pull request in? If you want, I could take care of this.

Reply:
Also, this check for image exists should probably go into the `handleImagePage` handler.

Reply:
Hmm. Okay, we could move it into handleImagePage, but my intention was to redirect back to the album page, if the album exists. Mostly I was trying to mirror what you did with albums -- if the album exists you call handleAlbumPage, but if it doesn't you return the 404 here in siteHandler.